src/profile-logic/profile-data.js

```diff
@@ -2125,9 +2126,34 @@ export function getThreadProcessDetails(thread: Thread): string {
   return label;
 }
 
+// Determines which information to show in the "origin annotation" if both an
+// origin name and a filename are present.
+// A return value of true means "show both", false means "only show filename."
+function _shouldShowBothOriginAndFileName(
+  fileName: string,
+  origin: string,
+  resourceType: resourceTypeEnum
+): boolean {
+  // If the origin string is just a URL prefix that's part of the
+  // filename, it doesn't add any useful information, so only show
+  // the filename.
+  if (fileName.startsWith(origin)) {
+    return false;
+  }
+
+  // For native code (resource type "library"), if we have the filename of the
+  // source code, only show the filename and not the library name.
+  if (resourceType === resourceTypes.library) {
+    return false;
+  }
```

Comment:
I wonder about that: isn't it useful to still see the library name in this case?

Reply:
It might be useful in some rare cases, but I couldn't really think of one. For Firefox profiling, the vast majority of interesting functions will just say "XUL" (or "libxul.so" or "xul.dll") which is mostly noise. And for users who really do need to know the library name, they can still right-click on the tree row and see the "collapse library" item in the context menu, which has the library name.